Screenbox/Pages/MainPage.xaml.cs

```diff
@@ -131,12 +134,55 @@ public void NavigateContent(Type pageType, object? parameter)
             ContentFrame.Navigate(pageType, parameter, new SuppressNavigationTransitionInfo());
         }
 
+        private void OnActualThemeChanged(FrameworkElement sender, object args)
+        {
+            UpdateSystemCaptionButton();
+        }
+
         private void SetTitleBar()
         {
             Window.Current.SetTitleBar(TitleBarElement);
             if (ApplicationView.GetForCurrentView()?.TitleBar is { } titleBar)
             {
+                // Restore colors upon returning to the page
                 titleBar.ButtonForegroundColor = null;
+                titleBar.ButtonHoverBackgroundColor = null;
+                titleBar.ButtonHoverForegroundColor = null;
+                titleBar.ButtonPressedBackgroundColor = null;
+                titleBar.ButtonPressedForegroundColor = null;
+                titleBar.ButtonInactiveForegroundColor = null;
+
+                UpdateSystemCaptionButton();
+            }
+        }
+
+        private void UpdateSystemCaptionButton()
+        {
+            var titleBar = ApplicationView.GetForCurrentView().TitleBar;
+            if (titleBar != null)
+            {
+                // Rest
+                var buttonBackgroundColor = (Color)Application.Current.Resources["CaptionButtonBackground"];
+                var buttonForegroundColor = (Color)Application.Current.Resources["CaptionButtonForeground"];
```

Comment:
Since the application theme never actually changes, you can't rely on application theme resources to get the color values for the active theme. If the system theme is Dark and the user changes the theme to Light in the app, these lines will still return values for Dark theme.